core/crypto/keccak/keccak.c

```diff
@@ -271,22 +245,23 @@ sha3_Finalize(void *priv)
 
   /* Return first bytes of the ctx->s. This conversion is not needed for
    * little-endian platforms e.g. wrap with #if !defined(__BYTE_ORDER__)
-   * || !defined(__ORDER_LITTLE_ENDIAN__) || __BYTE_ORDER__!=__ORDER_LITTLE_ENDIAN__
+   * || !defined(__ORDER_LITTLE_ENDIAN__) ||
+   * __BYTE_ORDER__!=__ORDER_LITTLE_ENDIAN__
    *    ... the conversion below ...
    * #endif */
   {
     unsigned i;
-    for(i = 0; i < SHA3_KECCAK_SPONGE_WORDS; i++) {
-      const unsigned t1 = (uint32_t) ctx->s[i];
-      const unsigned t2 = (uint32_t) ((ctx->s[i] >> 16) >> 16);
-      ctx->sb[i * 8 + 0] = (uint8_t) (t1);
-      ctx->sb[i * 8 + 1] = (uint8_t) (t1 >> 8);
-      ctx->sb[i * 8 + 2] = (uint8_t) (t1 >> 16);
-      ctx->sb[i * 8 + 3] = (uint8_t) (t1 >> 24);
-      ctx->sb[i * 8 + 4] = (uint8_t) (t2);
-      ctx->sb[i * 8 + 5] = (uint8_t) (t2 >> 8);
-      ctx->sb[i * 8 + 6] = (uint8_t) (t2 >> 16);
-      ctx->sb[i * 8 + 7] = (uint8_t) (t2 >> 24);
+    for (i = 0; i < SHA3_KECCAK_SPONGE_WORDS; i++) {  // NOLINT
```

Comment:
can you specify more precise what exactly lint rule should be avoided?

Reply:
this is external code, which should not be checked for clang-tidy

Reply:
actually the whole file should be excluded from clang-tidy, will specify this in cmake